### PR TITLE
Hide delete tag from case name

### PR DESCRIPTION
**Pull request by AmitPhulera**
Ticket - https://dimagi-dev.atlassian.net/browse/SAAS-10125
##### SUMMARY

- Corrected the condition which was checking if a given tag should be hidden.
- Reproduced the condition by injecting data directly to [instanceJSON](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/export/static/export/js/models.js#L79) variable as it was in HQ. As per my tests, the update resolves the ticket.

#### Screenshots
- On branch `ap/remove-delete-in-name`, which have recent changes, the deleted tag does not show up.
Screenshot 2020-04-08 at 6 21 33 PM

- On branch `master`, where deleted tag appears in name property.
Screenshot 2020-04-08 at 6 19 39 PM

### CODE REVIEW

**Comment by dannyroberts** [comment]
@AmitPhulera it's a little hard to follow just from the one line change—could you add before and after screenshots from your local testing to help me with that?

**Comment by AmitPhulera** [comment]
@dannyroberts I have added the screenshots in the PR description.

File: corehq/apps/export/static/export/js/models.js [hunk 1]
```diff
@@ -834,6 +837,15 @@ hqDefine('export/js/models', [
         return false;
     };
 
+    ExportColumn.prototype.removeDeletedTagFromCaseName = function () {
+      if(this.isCaseName() && this.is_deleted())
+      {
+        this.is_deleted(false);
+        var new_tags = this.tags().filter(ele => ele !== "deleted");
```
[the remark this answers is not in the document]
Comment (dannyroberts) [reply corehq/apps/export/static/export/js/models.js:844]: I might be behind the times, but I think our policy is still not to use full ES6 syntax in our JS code. @orangejenny can you confirm?

[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/export/static/export/js/models.js:844]: We don't have an official policy. I'd lean against using ES6 because it will break the linter and because we don't have any plans to switch over, so being consistent right now means using ES5. We don't have any plans to switch the linter.

[the remark this answers is not in the document]
Comment (AmitPhulera) [reply corehq/apps/export/static/export/js/models.js]: Okay. I have changed the function implementation to ES5.

**Review by dannyroberts — APPROVED** [review]
Looks good but one of the builds appears to be failing consistently: https://travis-ci.org/github/dimagi/commcare-hq/jobs/674774995. It's a JS test

**Comment by AmitPhulera** [comment]
The error that I see is in installation of PhantomJS.
I don't think it has to anything with the JS tests.

**Comment by dannyroberts** [comment]
I had restarted the build, but the previous error looked much more related. Sorry for the confusion. I just merged in master and we can see what happens with that build

File: corehq/apps/export/static/export/js/models.js [hunk 2]
```diff
@@ -923,7 +936,12 @@ hqDefine('export/js/models', [
      * @returns {Boolean} - True if it is the case name property False otherwise.
      */
     ExportItem.prototype.isCaseName = function () {
-        return this.path()[this.path().length - 1].name() === 'name';
+        try {
+            return this.path()[this.path().length - 1].name() === 'name';
+        } catch (error) {
+            console.log(error);
```
[the remark this answers is not in the document]
Comment (AmitPhulera) [reply corehq/apps/export/static/export/js/models.js:942]: Hey @orangejenny ,
Can you please guide me on what is the practice that we follow to handle/log exceptions on the front-end?
This one here is a rare occurrence, but tests were failing due to test data not having specific keys.

[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/export/static/export/js/models.js:942]: We don't log javascript errors. We've experimented with sending them to sentry, as described [here](https://sentry.io/answers/capture-errors/) under "Uncaught Exceptions" but haven't gotten that working (if you're interested, it's a good project, it isn't much code, it just hasn't been prioritized).

But for this particular function, I'd expect it to just return false if the path isn't formatted as expected. Would that work?

[the remark this answers is not in the document]
Comment (AmitPhulera) [reply corehq/apps/export/static/export/js/models.js:942]: Seems straight forward on the first look, I wonder if we can have it in coming sprints @dannyroberts ?

@orangejenny  Yeah just returning false would do fine for this case.

**Review by dannyroberts — APPROVED** [review]
